db/infinispan/src/main/java/org/commonjava/indy/infinispan/data/InfinispanStoreDataManager.java

```diff
@@ -139,9 +173,49 @@ public boolean isEmpty()
     }
 
     @Override
-    protected ArtifactStore putArtifactStoreInternal( StoreKey storeKey, ArtifactStore store )
+    protected synchronized ArtifactStore putArtifactStoreInternal( StoreKey storeKey, ArtifactStore store )
+    {
+        final ArtifactStore added = stores.put( storeKey, store );
+        final Map<StoreType, Set<StoreKey>> typedKeys =
+                storesByPkg.computeIfAbsent( storeKey.getPackageType(), k -> new HashMap<>() );
+        final Set<StoreKey> keys = typedKeys.computeIfAbsent( storeKey.getType(), k -> new HashSet<>() );
```

Comment:
Will this result in the new set being added to the map? I thought I got tripped up by this recently, where I needed putIfAbsent instead...

Reply:
It's a new set if there is no mapped set in this hashmap with the storeType here, or use the existed set from that type. I think computeIfAbsent is working this way IIRC.